### ISSUE TITLE
[Flow] Can't "import { type FormProps } from 'redux-form';"

### ISSUE THREAD

**Issue by ryami333** [erikras/redux-form#3823]
This appears to be because `Form.js.flow` is not published. In `index.js.flow` there are these relevant lines:
```jsx
import type { Props as _FormProps } from './Form'
...
export type FormProps = _FormProps
```

However `./Form` will not resolve to a `.flow` file, because one does not exist in `node_modules/redux_form/es` or `node_modules/redux-form/lib`.

This means that these docs are actually all incorrect: https://redux-form.com/7.3.0/docs/flow.md/. You can however workaround this by using this instead:
```
import { type FormProps } from 'redux-form/es/types.js.flow';
```

**Comment by ryami333** [comment]
@erikras this is becoming a bit of a blocker on a project I'm currently working on - would be happy to submit a PR but would be nice to hear your thoughts on this first because it seems to be a fundamental issue with the way types are exported/published in this library, and I could make a quick-fix for this isolated case but I wouldn't be confident that there wouldn't be any regression.

Also, there's already [an open PR](https://github.com/erikras/redux-form/pull/3820) for a simple Flow fix, and it's sat there unresolved now for about 3 weeks and I'd need some kind of indication that you'd be able to review/merge/deploy a bit more promptly than that.

**Comment by ryami333** [comment]
@erikras - final offer! Would you like help safely publishing flow types or no??

**Comment by gustavohenke** [comment]
Hi @ryami333, please submit us the PR if you can find the fix for it.
Then we can publish a patch version with it.

**Comment by ryami333** [comment]
It's not a 'fix' so much as a total overhaul is required to the way types are maintained and published in this package. This is why I'd like some engagement before I begin work on it, because it's non-trivial to implement. I've used `flow-copy-source` with success in a couple of other open-source libraries now - how would you feel about using this tool here too?

**Comment by ecnaidar** [comment]
@ryami333 also worth noting that `FormProps` that this package (probably) wants to export is not the one that is exported from `Form.js` but the one that is defined in `types.js.flow` 
https://github.com/erikras/redux-form/blob/master/src/types.js.flow#L93

**Comment by JalilArfaoui** [comment]
This issue should be renamed "Broken Flow types import" or something, because it's not specific to FormProps :cry:

**Comment by JalilArfaoui** [comment]
For `FieldProps`, the correct import is: 

```js
import { type FieldProps } from 'redux-form/es/FieldProps.types.js.flow'
```

**Comment by asazernik** [comment]
@JalilArfaoui Importing FieldProps works for me; FormProps does not, and the reason it doesn't is not applicable to FieldProps. I suspect there's something specific to your setup. What's the exact way in which it doesn't work? You're getting an incorrect type definition, or Flow is ignoring the type altogether?

**Comment by mike1808** [comment]
I think the author is planning to move to TypeScript because Flow types in this project just don't work very well. For example, if you wrap your component with any HOC (`reduxForm`, `valueMapper`) then all your component props are ignored which makes the whole point typing useless. I have to write my own types for HOC and other components. I want to join @ryami333 point of understanding whether this project wants to continue using Flow or it is better just to export them to `flow-typed` and maintain them there.

**Comment by ecnaidar** [comment]
@mike1808 TBH I think author is not planning to do anything active with this package, besides merging pull requests.
There is recent form project that seems to be more active   
https://github.com/final-form/final-form

**Comment by erikras** [comment]
Published fix in [`v8.2.0`](https://github.com/erikras/redux-form/releases/tag/v8.2.0).
